tests/packages/test_dependency.py

```diff
@@ -245,6 +245,17 @@ def test_complete_name() -> None:
             ["x"],
             "A[x] (>=1.6.5,!=1.8.0,<3.1.0)",
         ),
+        # test single version range exclusions
+        ("A", ">=1.8,!=2.0.*", None, "A (>=1.8,!=2.0.*)"),
+        ("A", ">=1.8,!=2.*", None, "A (>=1.8,!=2.*)"),
+        ("A", ">=1.8,!=2.*.*", None, "A (>=1.8,!=2.*)"),
+        # we verify that the range exclusion logic is not too eager
+        ("A", ">=1.8,<2.0 || >=2.1.0", None, "A (>=1.8,<2.0 || >=2.1.0)"),
```

Comment:
I removed the `assert` (see previous comment) and this test failed. However, the result `A (>=1.8,!=2.0.*)` seemed to be fine - even better than the expected result. Doesn't it? It seems that the assert fails because the precision differs. Maybe, this can be improved. (See next comment.)

However, since this is the only test that fails without the `assert`, I think a more clear one that fails without the assertion should be added at least, e.g. `("A", ">=1.8,<2.0 || >=2.2.0", None, "A (>=1.8,<2.0 || >=2.2.0)")`.